### PR TITLE
Create filter tag

### PR DESCRIPTION
**Pull request by ebabel-eu**
## Amsterdam styled components pull request

Before opening a pull request, please ensure:

- [ ] The default export from a file matches the file name
- [ ] You have been following the guidelines, written in the [README](../blob/master/docs/CONTRIBUTING.md#user-content-conventions-and-rules) file
- [ ] Your code has the necessary tests written
- [ ] You have updated the [CHANGELOG.md unreleased sections](../blob/master/CHANGELOG.md#muser-content-unreleased)

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: packages/asc-ui/src/components/FilterTag/FilterTag.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import { Close } from '@datapunt/asc-assets'
+import Tag, { Props as TagProps } from '../Tag/Tag'
+import FilterTagStyle from './FilterTagStyle'
+import Icon from '../Icon'
+
+export interface FilterTagProps extends TagProps {}
+
+const FilterTag: React.FC<FilterTagProps> = ({
+  filterTagOnClick,
```
Comment (royderks) [inline packages/asc-ui/src/components/FilterTag/FilterTag.tsx]: Would just call this `onClick` as that's how it's done for other components in the project as well

File: packages/asc-ui/src/components/FilterTag/FilterTagStyle.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { Props as TypographyProps } from '../Typography/TypographyStyle'
+import Typography from '../Typography'
+import { themeColor } from '../../utils'
+import IconStyle from '../Icon/IconStyle'
+
+export type Props = {} & TypographyProps & styledComponents.StyledProps<any>
+
+export const FilterTagStyleCSS = () => {
+  return css<Props>`
+    margin: 0 8px 8px 0;
```
Comment (royderks) [inline packages/asc-ui/src/components/FilterTag/FilterTagStyle.ts]: Have a look at the utility `themeSpacing` and please check with a designer if you need to use a different spacing than one that fits the 4 pixel grid

File: packages/asc-ui/src/components/FilterTag/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default } from './FilterTag'
```
Comment (royderks) [inline packages/asc-ui/src/components/FilterTag/index.ts]: You might want to export the style and props here as well

File: packages/asc-ui/src/components/FilterTag/FilterTag.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  children,
+  ...otherProps
+}) => {
+  const handleClick = (e: Event) => {
```
Comment (royderks) [inline packages/asc-ui/src/components/FilterTag/FilterTag.tsx]: I believe we look for mouse events and keyboard events specifically, look at thow this is done for other components like `Button`

File: packages/asc-ui/src/components/FilterTag/FilterTagStyle.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    font-size: 16px;
+    font-weight: 500;
+
+    &:hover {
+      background-color: ${themeColor('tint', 'level3')};
+      cursor: pointer;
+    }
+
+    ${IconStyle} {
+      margin: 0 5px;
```
Comment (chrisvanmook) [inline packages/asc-ui/src/components/FilterTag/FilterTagStyle.ts]: plz use the `themeSpacing` util here

File: tsconfig.json [hunk 6]
```diff
@@ -44,7 +44,7 @@
     "declarationMap": true,
     "strictNullChecks": true,
     "suppressImplicitAnyIndexErrors": true,
-    "noUnusedLocals": true,
+    "noUnusedLocals": false,
```
Comment (royderks) [inline tsconfig.json]: Not sure if this is the real solution as we have other components with onClick methods in stories...

File: packages/asc-ui/src/components/FilterTag/FilterTag.stories.tsx [hunk 7]
```diff
@@ -29,14 +30,27 @@ storiesOf('Atoms/FilterTag', module)
         ref: React.useRef(null),
       },
     ]
+
+    const filterTagHandleClick = (nr: number): void => {
+      filterTagElements = filterTagElements
```
Comment (chrisvanmook) [inline packages/asc-ui/src/components/FilterTag/FilterTag.stories.tsx]: Please don't mutate filterTagElements. Can u use a useState hook here?

File: packages/asc-ui/src/components/FilterTag/FilterTag.stories.tsx [hunk 8]
```diff
@@ -29,14 +30,27 @@ storiesOf('Atoms/FilterTag', module)
...
+        .filter(el => {
+          if (el.id !== nr) {
+            return el
+          }
+          return undefined
+        })
+        .map(el => el)
```
Comment (chrisvanmook) [inline packages/asc-ui/src/components/FilterTag/FilterTag.stories.tsx]: this seems like a useless .map...

File: packages/asc-ui/src/components/FilterTag/FilterTag.stories.tsx [hunk 9]
```diff
@@ -29,14 +30,27 @@ storiesOf('Atoms/FilterTag', module)
         ref: React.useRef(null),
       },
     ]
+
+    const filterTagHandleClick = (nr: number): void => {
+      filterTagElements = filterTagElements
+        .filter(el => {
```
Comment (chrisvanmook) [inline packages/asc-ui/src/components/FilterTag/FilterTag.stories.tsx]: imho, it's more elegant to use a shorthand statement here

File: packages/asc-ui/src/components/FilterTag/FilterTag.stories.tsx [hunk 10]
```diff
@@ -29,14 +30,27 @@ storiesOf('Atoms/FilterTag', module)
...
+        .map(el => el)
+    }
+
     return (
       <>
-        {myArray.map(element => (
+        {filterTagElements.map(element => (
           <FilterTag
-            ref={element.ref}
+            forwardRef={element.ref}
```
Comment (chrisvanmook) [inline packages/asc-ui/src/components/FilterTag/FilterTag.stories.tsx]: if we put filterTagElements elements in a state, and we update the state with the new active / filtered out elements, we don't need a ref / forwardRef here

**Comment by chrisvanmook** [comment]
FYI, you can add `noUnusedLocals: false,` to `.storybook/webpack.config.js` under awesome-typescript-plugin options

File: packages/asc-ui/src/components/FilterTag/FilterTag.tsx [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+import Icon from '../Icon'
+
+export interface FilterTagProps extends TagProps {}
+
+const FilterTag: React.FC<
+  FilterTagProps & React.ButtonHTMLAttributes<HTMLButtonElement>
+> = ({ children, ...otherProps }) => (
+  <FilterTagStyle $as="button" {...otherProps}>
+    {children}
+    <Icon inline size={14} color="primary">
```
[the remark this answers is not in the document]
Comment (ebabel-eu) [reply packages/asc-ui/src/components/FilterTag/FilterTag.tsx]: Beslising in standup: het helle FilterTag clickbaar te houden, niet alleen het cruisje

**Pull request merged**